packages/interactivity/src/test/utils.ts

```diff
@@ -25,4 +27,256 @@ describe( 'Interactivity API', () => {
+			function normalFn( x: number ) {
+				return x + 1;
+			}
+			const wrapped = withScope( normalFn );
+			expect( typeof wrapped ).toBe( 'function' );
+		} );
+
+		it( 'should call the original normal function', () => {
+			let called = false;
+			function normalFn( x: number ) {
+				called = true;
+				return x * 2;
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( normalFn ) );
+			const result = wrapped( 5 );
+
+			expect( result ).toBe( 10 );
+			expect( called ).toBe( true );
+			// After invocation, scope and namespace are reset.
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should set the scope and namespace and restore them afterwards', () => {
+			let called = false;
+			let scope: Scope;
+			let namespace: string;
+			function normalFn( x: number ) {
+				called = true;
+				scope = getScope();
+				namespace = getNamespace();
+				return x * 2;
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( normalFn ) );
+			const result = wrapped( 5 );
+
+			expect( result ).toBe( 10 );
+			expect( called ).toBe( true );
+			expect( scope! ).toBe( dummyScope );
+			expect( namespace! ).toBe( dummyNamespace );
+			// After invocation, scope and namespace are reset.
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should return an async function when passed a generator function', async () => {
+			function* gen() {
+				yield Promise.resolve( 'value' );
+				return 'done';
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( gen ) );
+			const resultPromise = wrapped();
+
+			expect( resultPromise ).toBeInstanceOf( Promise );
+
+			const result = await resultPromise;
+
+			expect( result ).toBe( 'done' );
+			// After invocation, scope and namespace are reset.
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should execute a generator function step by step maintaining scope and namespace', async () => {
+			const steps: Array< { scope: any; namespace: string } > = [];
+			function* gen() {
+				steps.push( { scope: getScope(), namespace: getNamespace() } );
+				const a = yield Promise.resolve( 1 );
+				steps.push( { scope: getScope(), namespace: getNamespace() } );
+				const b = yield Promise.resolve( a + 1 );
+				steps.push( { scope: getScope(), namespace: getNamespace() } );
+				return b * 2;
+			}
+			const callback = dummyScopeAndNS( () => withScope( gen ) );
+			const result = await callback();
+
+			expect( result ).toBe( 4 );
+			steps.forEach( ( step ) => {
+				expect( step.scope ).toBe( dummyScope );
+				expect( step.namespace ).toBe( dummyNamespace );
+			} );
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should yield correct values when promises resolve in generator functions', async () => {
```

Comment:
This test is redundant with the previous one (`should execute a generator function step by step maintaining scope and namespace`).